src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -155,24 +172,14 @@ public final ProxyHandler newProxyHandler() {
 	}
 
 	/**
-	 * Should proxy the given address
+	 * Returns true when the given {@link SocketAddress} should be reached via the configured proxy. When the method
+	 * returns false, the client should reach the address directly and bypass the proxy
 	 *
 	 * @param address the address to test
 	 * @return true if of type {@link InetSocketAddress} and hostname candidate to proxy
 	 */
 	public boolean shouldProxy(SocketAddress address) {
-		return address instanceof InetSocketAddress && shouldProxy(((InetSocketAddress) address).getHostString());
-	}
-
-	/**
-	 * Should proxy the given hostname
-	 *
-	 * @param hostName the hostname to test
-	 * @return true if candidate to proxy
-	 */
-	public boolean shouldProxy(@Nullable String hostName) {
-		return nonProxyHosts == null || hostName == null || !nonProxyHosts.matcher(hostName)
-		                                                                  .matches();
+		return address instanceof InetSocketAddress && !nonProxyHostPredicate.test(address);
```

Comment:
Can we keep the backwards compatibility with something like this?

```
 * @deprecated as of 0.9.10. Use {@link #shouldProxy(SocketAddress)}. This method will be removed in version 1.0.0.
 */
@Deprecated
public boolean shouldProxy(@Nullable String hostName) {
	return nonProxyHostPredicate == ALWAYS_PROXY || hostName == null ||
	        !((RegexShouldProxyPredicate) nonProxyHostPredicate).pattern.matcher(hostName)
	                                                                    .matches();
}
```

Reply:
makes sense!